WordPress/Classes/ViewRelated/NUX/LoginWPComViewController.swift

```diff
@@ -76,22 +144,99 @@ class LoginWPComViewController: SigninWPComViewController, LoginViewController {
             return
         }
 
-        // If the username is not reserved proceed with the signin
-        if SigninHelpers.isUsernameReserved(loginFields.username) {
-            handleReservedUsername(loginFields.username)
-            return
-        }
-
         configureViewLoading(true)
 
         loginFacade.signIn(with: loginFields)
     }
 
-    override func displayLoginMessage(_ message: String!) {
-        // no-op
+    // Update safari stored credentials. Call after a successful sign in.
+    ///
+    func updateSafariCredentialsIfNeeded() {
+        SigninHelpers.updateSafariCredentialsIfNeeded(loginFields)
+    }
+
+    // MARK: - Actions
+
+    @IBAction func handleTextFieldDidChange(_ sender: UITextField) {
+        guard let usernameField = usernameField,
+            let passwordField = passwordField else {
+                return
+        }
+
+        loginFields.username = usernameField.nonNilTrimmedText()
+        loginFields.password = passwordField.nonNilTrimmedText()
+
+        configureSubmitButton(animating: false)
     }
 
-    override func displayRemoteError(_ error: Error!) {
+    @IBAction func handleSubmitButtonTapped(_ sender: UIButton) {
+        validateForm()
+    }
+
+    @IBAction func handleForgotPasswordButtonTapped(_ sender: UIButton) {
+        SigninHelpers.openForgotPasswordURL(loginFields)
+    }
+
+    func handleOnePasswordButtonTapped(_ sender: UIButton) {
+        view.endEditing(true)
+
+        SigninHelpers.fetchOnePasswordCredentials(self, sourceView: sender, loginFields: loginFields) { [weak self] (loginFields) in
+            self?.usernameField?.text = loginFields.username
+            self?.passwordField?.text = loginFields.password
+            self?.validateForm()
+        }
+    }
+
+    // MARK: - Keyboard Notifications
+
+    func handleKeyboardWillShow(_ notification: Foundation.Notification) {
+        keyboardWillShow(notification)
+    }
+
+    func handleKeyboardWillHide(_ notification: Foundation.Notification) {
+        keyboardWillHide(notification)
+    }
+
+    override func dismiss() {
+        self.performSegue(withIdentifier: .showEpilogue, sender: self)
```

Comment:
Suggestion here.  Maybe rename this to `showEpilogue` or some such.  We might want to keep the original dismiss implementation intact in order to dismiss the controller if, for instance, a user who already has self-hosted blogs in the app decides to cancel the wpcom login flow.  Just a thought.